runtime/src/bank_forks.rs

```diff
@@ -942,10 +970,26 @@ mod tests {
         let mut banks = vec![Arc::new(Bank::new_for_tests(&genesis_config))];
         assert_eq!(banks[0].slot(), 0);
         let mut bank_forks = BankForks::new_from_banks(&banks, 0);
-        banks.push(bank_forks.insert(Bank::new_from_parent(&banks[0], &Pubkey::default(), 1)));
-        banks.push(bank_forks.insert(Bank::new_from_parent(&banks[1], &Pubkey::default(), 2)));
-        banks.push(bank_forks.insert(Bank::new_from_parent(&banks[0], &Pubkey::default(), 3)));
-        banks.push(bank_forks.insert(Bank::new_from_parent(&banks[3], &Pubkey::default(), 4)));
+        banks.push(bank_forks.insert(Bank::new_from_parent(
+            banks[0].clone(),
+            &Pubkey::default(),
+            1,
+        )));
+        banks.push(bank_forks.insert(Bank::new_from_parent(
+            banks[1].clone(),
+            &Pubkey::default(),
+            2,
+        )));
+        banks.push(bank_forks.insert(Bank::new_from_parent(
+            banks[0].clone(),
+            &Pubkey::default(),
+            3,
+        )));
+        banks.push(bank_forks.insert(Bank::new_from_parent(
+            banks[3].clone(),
+            &Pubkey::default(),
+            4,
+        )));
```

Comment:
follow-up: this seems like a good use of a loop

Reply:
https://github.com/solana-labs/solana/pull/32852